src/views/PositionPdfPlot/PositionPdfPlotWidget.tsx

```diff
@@ -0,0 +1,229 @@
+import { useRecordingSelectionTimeInitialization, useTimeRange } from 'contexts/RecordingSelectionContext'
+import { matrix, multiply } from 'mathjs'
+import React, { FunctionComponent, useCallback, useMemo } from 'react'
+import TimeScrollView, { TimeScrollViewPanel, use1dTimeToPixelMatrix, usePanelDimensions, usePixelsPerSecond } from '../RasterPlot/TimeScrollView/TimeScrollView'
+import useFetchCache from './useFetchCache'
+
+export type FetchSegmentQuery = {
+    type: 'fetchSegment'
+    segmentNumber: number
+    segmentSize: number
+    downsampleFactor: number
+}
+
+type Props = {
+    fetchSegment: (query: FetchSegmentQuery) => Promise<number[][]>
+    startTimeSec: number
+    endTimeSec: number
+    samplingFrequency: number
+    numPositions: number
+    segmentSize: number
+    multiscaleFactor: number
+    width: number
+    height: number
+}
+
+type PanelProps = {
+}
+
+const margins = {
+    left: 30,
+    right: 20,
+    top: 20,
+    bottom: 50
+}
+
+const panelSpacing = 4
+
+const usePositionPdfDataModel = (fetchSegment: (q: FetchSegmentQuery) => Promise<number[][]>, numTimepoints: number, numPositions: number, segmentSize: number, multiscaleFactor: number) => {
+    const fetchSegmentCache = useFetchCache<FetchSegmentQuery>(fetchSegment)
+    const get = useCallback((i1: number, i2: number, downsampleFactor: number) => {
+        const s1 = Math.floor(i1 / segmentSize)
+        const s2 = Math.ceil(i2 / segmentSize)
+        if (s2 - s1 + 1 >= 8) return undefined
+        // if (downsampleFactor > 9) return undefined
+        const ret = allocate2d(i2 - i1, numPositions, undefined)
+        for (let s = s1; s < s2; s++) {
+            const S = fetchSegmentCache.get({type: 'fetchSegment', segmentNumber: s, segmentSize, downsampleFactor})
+            if (S) {
```

Comment:
I think my suggested alterations to the downsampling loop in `PositionPdfPlotView` might be applicable here too.